### PR TITLE
DEPR: deprecate msgpack support

### LINKED ISSUE #27084: DEPR: msgpack
msgpack uses the internal format on the wire. this makes for *only* pandas producers and consumers, but with ``pyarrow`` support now fully handling this case (and better performing), it is time to deprecate msgpack.

### PR DESCRIPTION
**Pull request by jreback**
closes #27084

### CODE REVIEW

File: pandas/io/packers.py [hunk 1]
```diff
@@ -92,6 +91,12 @@ def to_msgpack(path_or_buf, *args, **kwargs):
     compress : type of compressor (zlib or blosc), default to None (no
                compression)
     """
+    warnings.warn("The read_msgpack is deprecated and will be removed in a "
```
Comment (jbrockmendel) [inline pandas/io/packers.py]: "The " probably doesnt belong.

File: pandas/tests/io/test_packers.py [hunk 2]
```diff
@@ -715,14 +728,18 @@ def test_compression_warns_when_decompress_caches_blosc(self, monkeypatch):
     def _test_small_strings_no_warn(self, compress):
         empty = np.array([], dtype='uint8')
         with tm.assert_produces_warning(None):
-            empty_unpacked = self.encode_decode(empty, compress=compress)
+            with catch_warnings():
+                filterwarnings('ignore', category=FutureWarning)
```
Comment (jbrockmendel) [inline pandas/tests/io/test_packers.py]: how does this interact with the tm.assert_produces_warning above?

**Comment by jbrockmendel** [comment]
I’m generally on board with this, but concerned about the fact that msgpack uses internals, so if this becomes “downstream” then that may make it harder to simplify things in internals if we are worried about breaking downstream.

**Comment by jreback** [comment]
> I’m generally on board with this, but concerned about the fact that msgpack uses internals, so if this becomes “downstream” then that may make it harder to simplify things in internals if we are worried about breaking downstream.

that’s all the more reason to deprecate this actually.

**Pull request merged**